Cython/Build/SharedModule.py

```diff
@@ -0,0 +1,74 @@
+import tempfile
+import os
+import shutil
+
+from Cython.Compiler import (
+    MemoryView, Code, Options, Pipeline, Errors, Main, Symtab
+)
+from Cython.Compiler.StringEncoding import EncodedString
+from Cython.Compiler.Scanning import FileSourceDescriptor
+
+def create_shared_library_pipeline(context, scope, options, result):
+
+    def generate_tree_factory(context):
+        def generate_tree(compsrc):
+
+            # Force to generate __Pyx_ExportFunction
+            Options.cimport_from_pyx = True
```

Comment:
As a simple mitigation for this global setting, we should at least reset the original value after our use.